### PR TITLE
fix(import): duplicate properties are emitted when they differ only by case

### LINKED ISSUE #2341: Unable to import CRD from VictoriaMetrics operator
### Description of the bug:

Unable to import [CRD](https://github.com/VictoriaMetrics/operator/releases/download/v0.39.1/crd.yaml) from VictoriaMetric Operator 

### Reproduction Steps:

`cdk8s import https://github.com/VictoriaMetrics/operator/releases/download/v0.39.1/crd.yaml --language python `

### Error Log:

Its full of error like below

```
operator.victoriametrics.com
  operator.victoriametrics.com/vmsingle
NOTE: Temp directory retained due to an error: /tmp/temp-F9LVQF
Error: jsii compilation failed with non-zero exit code: 1
  | [2023-11-06T16:24:03.363] [ERROR] jsii/compiler - Compilation errors prevented the JSII assembly from being created
  | warning JSII6: A "peerDependency" on "constructs" at "10.3.0" means you should take a "devDependency" on "constructs" at "10.3.0" (found "undefined")
  | warning JSII6: A "peerDependency" on "cdk8s" at "2.67.6" means you should take a "devDependency" on "cdk8s" at "2.67.6" (found "undefined")
  | warning JSII3: There is no "README.md" file. It is required in order to generate valid PyPI (Python) packages.
  | com.victoriametrics.operator.ts:2218:12 - error TS2300: Duplicate identifier 'sourceLabels'.
  | 2218   readonly sourceLabels?: string[];
  |                 ~~~~~~~~~~~~
  | com.victoriametrics.operator.ts:2225:12 - error TS2300: Duplicate identifier 'sourceLabels'.
  | 2225   readonly sourceLabels?: string[];
  |                 ~~~~~~~~~~~~
  | com.victoriametrics.operator.ts:2232:12 - error TS2300: Duplicate identifier 'targetLabel'.
  | 2232   readonly targetLabel?: string;
  |                 ~~~~~~~~~~~
  | com.victoriametrics.operator.ts:2239:12 - error TS2300: Duplicate identifier 'targetLabel'.
  | 2239   readonly targetLabel?: string;
  |                 ~~~~~~~~~~~
  | com.victoriametrics.operator.ts:2338:12 - error TS2300: Duplicate identifier 'sourceLabels'.
  | 2338   readonly sourceLabels?: string[];
  |                 ~~~~~~~~~~~~
  | com.victoriametrics.operator.ts:2345:12 - error TS2300: Duplicate identifier 'sourceLabels'.
  | 2345   readonly sourceLabels?: string[];
  |                 ~~~~~~~~~~~~
  | com.victoriametrics.operator.ts:2352:12 - error TS2300: Duplicate identifier 'targetLabel'.
  | 2352   readonly targetLabel?: string;
  |                 ~~~~~~~~~~~
  | com.victoriametrics.operator.ts:2359:12 - error TS2300: Duplicate identifier 'targetLabel'.
  | 2359   readonly targetLabel?: string;
  |                 ~~~~~~~~~~~
  +----------------------------------------------------------------------------------
  | Command: /usr/lib/node_modules/cdk8s-cli/node_modules/jsii-srcmak/node_modules/jsii/bin/jsii --silence-warnings reserved-word
  | Workdir: /tmp/temp-F9LVQF
  +----------------------------------------------------------------------------------
```

### Environment:

  - **Framework Version**: cdk8s 2.163.0
  - **OS**: debian bookworm

### Other:

I have also raised an issue in VictoriaMetrics [here](https://github.com/VictoriaMetrics/operator/issues/804)

---

This is :bug: Bug Report

### LINKED ISSUE #2722: `import --typescript` generates duplicate interface property, when source uses differing case (i.g. snack_case and camelCase)
### Description of the bug:

After running the CDK8s import command to generate types for the [Postgres-Operator's `postgresql.yaml` manifest](https://github.com/zalando/postgres-operator/blob/master/manifests/postgresql.crd.yaml), it generated a few duplicate types. For example `podPriorityClassName` was duplicated.

i.g.

```ts
 /**
   * deprecated
   *
   * @schema PostgresqlSpec#pod_priority_class_name
   */
  readonly podPriorityClassName?: string;

  /**
   * @schema PostgresqlSpec#podPriorityClassName
   */
  readonly podPriorityClassName?: string;
```

It looks like this can be reproduced when the imported file has duplicate property names in different casing styles.

Here's what the import command is trying to translate:

```yaml
pod_priority_class_name:
    type: string
    description: deprecated
podPriorityClassName:
    type: string
```

Link to source manifest code:
https://github.com/zalando/postgres-operator/blob/7d4da928726b5029f72e9ab83ee9b6ff481e70c7/manifests/postgresql.crd.yaml#L353-L357

### Reproduction Steps:

Can be reproduced using the `postgresql.crd.yaml` CRD from [Postgres-Operator](https://github.com/zalando/postgres-operator/blob/master/manifests/postgresql.crd.yaml)

```sh
curl https://raw.githubusercontent.com/zalando/postgres-operator/7d4da928726b5029f72e9ab83ee9b6ff481e70c7/manifests/postgresql.crd.yaml -o ./postgresql.crd.yaml
```

Then run the import command:

```sh
cdk8s import --language typescript postgresql:=./postgresql.crd.yaml
# outputs postgresql-acid.zalan.do.ts
```

Then open the generated file at `postgresql-acid.zalan.do.ts` and search for `podPriorityClassName` (line 217) which belongs to the `PostgresqlSpec` interface. Notice how it's right next to a duplicate definition.

### Error Log:

When running `tsc`:
```sh
src/imports/postgresql-acid.zalan.do.ts:217:12 - error TS2300: Duplicate identifier 'podPriorityClassName'.

217   readonly podPriorityClassName?: string;
               ~~~~~~~~~~~~~~~~~~~~
src/imports/postgresql-acid.zalan.do.ts:222:12 - error TS2300: Duplicate identifier 'podPriorityClassName'.

222   readonly podPriorityClassName?: string;
               ~~~~~~~~~~~~~~~~~~~~
```

### Environment:

  - **Framework Version**: CDK8s CLI: v2.0.13
  - **OS**: Monterey v12.4

### Other:

---

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by iliapolo**
Fixes https://github.com/cdk8s-team/cdk8s-cli/issues/2341
Fixes https://github.com/cdk8s-team/cdk8s-cli/issues/2722

----

This requires merging https://github.com/cdklabs/json2jsii/pull/1560 and pulling in the new version of `json2jsii`.

### CODE REVIEW

File: .projenrc.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ const project = new Cdk8sTeamTypeScriptProject({
     'sscaff',
     'yaml',
     'yargs@^15',
-    'json2jsii@0.4.5',
+    'json2jsii@0.5.0',
```
Comment (iliapolo) [inline .projenrc.ts:49]: This is the fix

File: test/import/fixtures/postgresql.yaml [hunk 2]
```diff
@@ -0,0 +1,645 @@
...
+                    type: integer
+              podAnnotations:
+                type: object
+                additionalProperties:
+                  type: string
+              pod_priority_class_name:
+                type: string
+                description: deprecated
+              podPriorityClassName:
+                type: string
```
Comment (iliapolo) [inline test/import/fixtures/postgresql.yaml:357]: This is the culprit

File: test/import/__snapshots__/import-crd.test.ts.snap [hunk 3]
```diff
@@ -123892,6 +123892,5645 @@ export function toJson_TenantV1Beta1SpecNetworkPoliciesItemsIngressFromPodSelect
...
+   * @schema PostgresqlSpec#podAnnotations
+   */
+  readonly podAnnotations?: { [key: string]: string };
+
+  /**
+   * deprecated
+   *
+   * @schema PostgresqlSpec#pod_priority_class_name
+   */
+  readonly podPriorityClassName?: string;
```
Comment (iliapolo) [inline test/import/__snapshots__/import-crd.test.ts.snap:128072]: This is what we are asserting again. Only one property is emitted, like expected.

File: test/import/__snapshots__/import-crd.test.ts.snap
Comment (iliapolo) [inline test/import/__snapshots__/import-crd.test.ts.snap:1]: Note this file only has additions, so no breaking changes.

**Review by rix0rrr — APPROVED** [review]

**Review by mrgrain — APPROVED** [review]

**Pull request merged**